metaflow/plugins/kubernetes/kube_cli.py

```diff
@@ -0,0 +1,276 @@
+
+
+@kube.command(help="List running Kube tasks of this flow")
+@click.option(
+    "--my-runs", default=False, is_flag=True, help="Run the command over all tasks."
+)
+@click.option("--user", default=None, help="List tasks for the given user.")
+@click.option("--run-id", default=None, help="List tasks corresponding to the run id.")
+@click.pass_context
+def list(ctx, run_id, user, my_runs):
+    batch = Kube(ctx.obj.metadata, ctx.obj.environment,ctx.obj.datastore)
+    _execute_cmd(
+        batch.list_jobs, ctx.obj.flow.name, run_id, user, my_runs, ctx.obj.echo
+    )
+
+
+@kube.command(help="Terminate running Kube tasks of this flow.")
+@click.option("--my-runs", default=False, is_flag=True, help="Kill all running tasks.")
+@click.option("--user", default=None, help="List tasks for the given user.")
+@click.option(
+    "--run-id", default=None, help="Terminate tasks corresponding to the run id."
+)
+@click.pass_context
+def kill(ctx, run_id, user, my_runs):
+    batch = Kube(ctx.obj.metadata, ctx.obj.environment,ctx.obj.datastore)
+    _execute_cmd(
+        batch.kill_jobs, ctx.obj.flow.name, run_id, user, my_runs, ctx.obj.echo
+    )
+
+
+@kube.command(
+    help="Execute a single task using Kube. This command "
+    "calls the top-level step command inside a Kube Job Container"
+    "with the given options. Typically you do not "
+    "call this command directly; it is used internally "
+    "by Metaflow."
+)
+@click.argument("step-name")
+@click.argument("code-package-sha")
+@click.argument("code-package-url")
+@click.option("--executable", help="Executable requirement for Kube.")
+@click.option(
+    "--image", help="Docker image requirement for Kube. In name:version format."
+)
+@click.option("--cpu", help="CPU requirement for Kube.")
+@click.option("--gpu", help="GPU requirement for Kube.")
+@click.option("--memory", help="Memory requirement for Kube.")
+@click.option("--run-id", help="Passed to the top-level 'step'.")
+@click.option("--task-id", help="Passed to the top-level 'step'.")
+@click.option("--input-paths", help="Passed to the top-level 'step'.")
+@click.option("--split-index", help="Passed to the top-level 'step'.")
+@click.option("--clone-path", help="Passed to the top-level 'step'.")
+@click.option("--clone-run-id", help="Passed to the top-level 'step'.")
+@click.option(
+    "--tag", multiple=True, default=None, help="Passed to the top-level 'step'."
+)
+@click.option("--namespace", default=None, help="Passed to the top-level 'step'.")
+@click.option("--kube_namespace", default=None, help="Name Space to use on Kubernetes. This is not a metaflow namespace")
+@click.option("--retry-count", default=0, help="Passed to the top-level 'step'.")
+@click.option(
+    "--max-user-code-retries", default=0, help="Passed to the top-level 'step'."
+)
+@click.option(
+    "--run-time-limit",
+    default=5 * 24 * 60 * 60,
+    help="Run time limit in seconds for the Batch job. " "Default is 5 days.",
+)
+@click.pass_context
+def step(
```

Comment:
This function specifically and also the file overall look extremely similar to batch_cli.py. It could make sense to refactor everything to get rid of code duplication and avoid future double maintenance.